wallet-new/src/Cardano/Wallet/Kernel/DB/Spec/Update.hs

```diff
@@ -85,16 +86,25 @@ instance Arbitrary NewForeignFailed where
     arbitrary = pure . NewForeignInputsAvailable . InDb $ mempty
 
 -- | Errors thrown by 'applyBlock'
-data ApplyBlockFailed =
-    -- | The block we're trying to apply does not fit onto the previous
-    --
-    -- This indicates that the wallet has fallen behind the node (for example,
-    -- when the node informs the wallet of a block but the wallet gets
-    -- shut down before it gets a chance to process it).
-    --
-    -- We record  the context of the block we're trying to apply and the
-    -- context of the most recent checkpoint.
-    ApplyBlockNotSuccessor BlockContext (Maybe BlockContext)
+data ApplyBlockFailed
```

Comment:
Let's not modify the spec error type, but rather introduce a _new_ error type that has the 3 constructors you introduced. This way we make it clear that the original error cannot propagate up.